### PR TITLE
Handle SSL handshake exception on SCM callback step

### PR DESCRIPTION
**Pull request by vinokurig**
### What does this PR do?
Intercept the SSLHandshakeException and pass the `ssl_exception` error code when redirecting to dashboard on oAuth callback call.

### What issues does this PR fix or reference?

https://issues.redhat.com/browse/CRW-6940

### How to test this PR?

1. Deploy Che from the pull request image: `quay.io/eclipse/che-server:pr-710`
2. Apply the related dashboard image: `quay.io/eclipse/che-dashboard:pr-1171`
3. Configure oAuth for a self hosted git provider that uses self signed certificate e.g. https://gh.crw-qe.com/
4. Start a workspace from the provider's repository with a devfile.

See: A notification appears in the dashboard side:
screenshot-eclipse-che apps rosa tnx7p-xgwvy-6er z6p6 p3 openshiftapps com-2024 08 20-15_02_22

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java [hunk 1]
```diff
@@ -117,32 +118,39 @@ public Response callback(UriInfo uriInfo, @Nullable List<String> errorValues)
...
       return Response.temporaryRedirect(
-              URI.create(
-                  getParameter(params, "redirect_after_login")
-                      + String.format("&%s=access_denied", ERROR_QUERY_NAME)))
+              URI.create(getRedirectAfterLoginUrl(params, "access_denied")))
           .build();
     } catch (UnsatisfiedScmPreconditionException | ScmConfigurationPersistenceException e) {
       // Skip exception, the token will be stored in the next request.
       LOG.error(e.getMessage(), e);
+    } catch (ScmCommunicationException e) {
```
Comment (tolusha) [inline wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:126]: I don't quite understand why ScmCommunicationException is treated like ssl problem.
Could you explain please?

Comment (vinokurig) [reply wsmaster/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/EmbeddedOAuthAPI.java:127]: That's a good point, I added a check by error code.

**Review by tolusha — APPROVED** [review]

**Comment by artaleks9** [comment]
Verified on Eclipse Che with `quay.io/eclipse/che-dashboard:pr-1171` and `quay.io/eclipse/che-server:pr-710`
Checked using gitlab server with SSl certificate https://gitlab-gitlab-system.apps.git.crw-qe.com

**The functionality works properly**
Eclipse-Che-Creating-a-workspace-pr-710

**Pull request merged**